### DISCUSSION TITLE
收不到 Telegram 通知

### DISCUSSION THREAD

**Discussion by iamgodot** [walinejs/waline#644] [Q&A] [answered]
按照官方教程基于 Leancloud 和 Vercel 部署的，在设置 Telegram 之后发评论一直收不到通知。

1. 环境变量只设置了必填的 TG_BOT_TOKEN 和 TG_CHAT_ID
2. 设置完之后重新部署过，没有使用 cache
3. 单独测试 Telegram 的 bot api 是可以发送消息的

在 Vercel 的界面一直看不到任何 log，所以不知道如何继续排查了，求助

**Comment by iamgodot** [comment]
又重新部署了下，可以看到日志了，但是评论之后并没有看到推送 Telegram 消息的日志，是没有相关 log 吗

**Reply by lizheming** [reply]
嗯，tg 发送的逻辑在这里 https://github.com/walinejs/waline/blob/main/packages/server/src/service/notify.js#L189-L198 目前没有加相关的日志

**Reply by iamgodot** [reply]
> 嗯，tg 发送的逻辑在这里 https://github.com/walinejs/waline/blob/main/packages/server/src/service/notify.js#L189-L198 目前没有加相关的日志

谢谢，这段代码我看过应该不是问题，所以如果没有报错日志的话还有什么别的原因会造成没有触发 TG 的消息发送吗？

**Comment by iamgodot** [comment, marked as answer]
在 Vercel 上面重新创建了实例之后成功了，看起来现在 TG 对 md 的支持不是很友好，比如 self.link 会因为 `.` 解析出错的
